cmd/start.go

```diff
@@ -22,12 +28,25 @@ Options passed on start override configuration options only on start and are not
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
 
-			store, err := nodebuilder.OpenStore(StorePath(ctx))
+			// override config with all modifiers passed on start
+			cfg := NodeConfig(ctx)
+
+			storePath := StorePath(ctx)
+			keysPath := filepath.Join(storePath, "keys")
+
+			// construct ring
+			// TODO @renaynay: Include option for setting custom `userInput` parameter with
+			//  implementation of https://github.com/celestiaorg/celestia-node/issues/415.
+			encConf := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+			ring, err := keyring.New(app.Name, cfg.State.KeyringBackend, keysPath, os.Stdin, encConf.Codec)
+			if err != nil {
+				return err
+			}
+
+			store, err := nodebuilder.OpenStore(storePath, ring)
```

Comment:
I expected that OpenStore would open the keyring inside because opening does it for every piece of data presented in the folder. Do you have any reason to keep it outside?

Reply:
The main reason is that we need config information for "opening" the keyring and i'd rather not pass the config to `OpenStore` since it's unrelated - unless we make some store config and store the paths and other params in there.